4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+    select(date, month, year, doy, decade, period, gcm, ice, hice, evap, n_layers, everything()) # Place the temperature prediction columns last
+  
+  return(site_preds)
+}
+
+#' @title Get surface, middle, and bottom GLM predictions for all lakes
+#' @description Function to pull and combine GLM predictions for all lakes,
+#' across all 3 simulation periods and all 6 GCMs
+#' @param run_groups The `p2_glm_uncalibrated_run_groups` grouped version 
+#' of the `p2_glm_uncalibrated_runs` output tibble subset to the site_id, 
+#' gcm, time_period, raw_meteo_fl, export_fl, and export_fl_hash columns 
+#' and grouped by site_id and gcm. Then filtered to only groups for which 
+#' glm_success==TRUE for all runs in that group.
+#' @return A munged dataframe of surface, middle, and bottom temperature GLM
+#' predictions for all successfully simulated lakes. The munge table includes 
+#' the columns site_id, gcm, date, month, year DOY, decade, period, gcm, 
+#' ice, hice (ice thickness), evap (evaporation), n_layers (number of simulated 
+#' layers), depth_class (surface, middle, or bottom), depth, and temperature 
+get_surface_middle_bottom_preds <- function(run_groups) {
+  surface_middle_bottom_preds <- run_groups %>%
+    group_by(site_id) %>%
+    group_modify(~ {
+      site_meteo_files <- .x$raw_meteo_fl
+      site_export_files <- .x$export_fl
+      run_gcm_list <- .x$gcm
+      # Pull predictions for each site
+      site_preds <- get_site_preds(site_meteo_files, site_export_files, run_gcm_list)
+      
+      # Munge to long format
+      site_preds_long <- munge_long(site_preds)
+      
+      # Identify surface, middle, and bottom depths for each site and filter
+      # predictions to only those depths
+      plot_depths <- get_plotting_depths(site_preds_long)
+      filtered_preds <- filter(site_preds_long, depth %in% plot_depths) %>%
+        mutate(depth_class=case_when(
+          depth==plot_depths$plot_depth_surface ~ 'surface',
+          depth==plot_depths$plot_depth_middle ~ 'middle',
+          depth==plot_depths$plot_depth_bottom ~ 'bottom'
+        ), .before=depth)
+      return(filtered_preds)
+    })
+  return(surface_middle_bottom_preds)
+}
+
+#' @title Munge GLM predictions from wide to long format
+#' @description pivots wide-format GLM predictions (with columns like
+#' 'temp_0', 'temp_0.5', etc.) to wide format, with a column for 'depth'
+#' and a column for 'temperature'
+#' @param input_wide The wide-formatted input data
+#' @return a munged dataframe of GLM predictions that includes all of 
+#' the columns in the original wide data except the 'temp_X' columns,
+#' which are instead reduced to two columns: 'depth' and 'temperature'
+munge_long <- function(input_wide) {
+  input_long <- input_wide %>%
+    pivot_longer(temp_0:colnames(input_wide)[ncol(input_wide)], names_to="depth", values_to="temperature") %>%
```

Comment:
I think you just want to pivot the temp_* columns, right? You could change this to use a dplyr helper function to be a bit more robust

```suggestion
    pivot_longer(starts_with("temp_"), names_to="depth", values_to="temperature") %>%
```

Reply:
oooh did not know you could use that helper in this way - thanks!